src/Aspire.Hosting/Dashboard/DashboardTokenProvider.cs

```diff
@@ -0,0 +1,43 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Text;
+using Aspire.Hosting.Utils;
+using Microsoft.Extensions.Options;
+
+namespace Aspire.Hosting.Dashboard;
+
+internal interface IDashboardTokenProvider
+{
+    public string BrowserToken { get; }
+    public string OltpToken { get; }
+    public string ResourceServerToken { get; }
+}
+
+internal class DashboardTokenProvider : IDashboardTokenProvider
+{
+    public DashboardTokenProvider(IOptions<TransportOptions> transportOptions)
+    {
+        BrowserToken = transportOptions.Value.BrowserToken ?? GenerateToken();
```

Comment:
Add debug level logging here about whether the token from the options are used or one is generated.